lib/amqp/session.rb

```diff
@@ -930,9 +930,15 @@ def receive_frame(frame)
         begin
           receive_frameset(@frames[frame.channel])
         ensure
-          # for channel.close, frame.channel will be nil. MK.
-          clear_frames_on(frame.channel) if @frames[frame.channel]
+          begin
+            clear_frames_on(frame.channel) if @frames[frame.channel] # for channel.close, frame.channel will be nil. MK.
+          rescue Exception => ex
```

Comment:
Really?  https://github.com/Invoca/exception_handling/blob/master/lib/exception_handling/methods.rb#L27-L34
Does Ringswitch have its own copy of `ensure_safe`?!